### PR TITLE
support setting config after creating http proxy handler

### PR DESCRIPTION
**Pull request by wangrzneu**
support setting config after creating http proxy handler

Signed-off-by: renzheng.wang <renzheng.wang@ucloud.cn>

### CODE REVIEW

File: proxy/httpproxy/rawkv.go [hunk 1]
```diff
@@ -49,7 +50,13 @@ type RawResponse struct {
 }
 
 func (h rawkvHandler) New(ctx context.Context, r *RawRequest) (*RawResponse, int, error) {
-	id, err := h.p.New(ctx, r.PDAddrs, config.Default())
+	var id proxy.UUID
+	var err error
+	if h.c == nil {
```
Comment (disksing) [inline proxy/httpproxy/rawkv.go]: It seems it will never be nil.

**Comment by disksing** [comment]
Thanks @wangrzneu . Your PR will fix https://github.com/tikv/client-go/issues/18 after get merged.

File: proxy/httpproxy/rawkv.go (hunk 1 above)
Comment (wangrzneu) [reply proxy/httpproxy/rawkv.go]: I think this check can make the code safe, although h.c cannot be nil for current usage of rawkvHandler.

**Comment by disksing** [comment]
I think you did not add the method to allow client to update the config...

File: proxy/httpproxy/rawkv.go (hunk 1 above)
Comment (disksing) [reply proxy/httpproxy/rawkv.go]: em, how about add a `getConfig()` method to `rawkvHandler`, inside the method, when c is nil, return the default config. In this way, you will not have to pay extra attention in `newHTTPProxyHandler`.

**Comment by wangrzneu** [comment]
The config can be updated after creating HTTPProxyHandler

**Comment by wangrzneu** [comment]
Do we need to update the config through the http calling?

File: proxy/httpproxy/txnkv.go [hunk 2]
```diff
@@ -269,3 +270,10 @@ func (h txnkvHandler) handlerFunc(f func(context.Context, *TxnRequest) (*TxnResp
 		w.Write(data)
 	}
 }
+
+func (h txnkvHandler) getConfig() config.Config {
+	if h.c == nil {
+		return config.Default()
+	}
+	return *h.c
+}
```
Comment (disksing) [inline proxy/httpproxy/txnkv.go]: please end the file with an empty new line.

**Comment by disksing** [comment]
I think I missunderstood your intention.. #18 means to update config by HTTP request. But I think we can merge your PR then handle #18 later.

**Comment by wangrzneu** [comment]
> I think I missunderstood your intention.. #18 means to update config by HTTP request. But I think we can merge your PR then handle #18 later.

I can propose a PR after current PR is merged.

**Comment by disksing** [comment]
Thanks for your contribution! @wangrzneu

**Pull request merged**